universal-application-tool-0.0.1/app/controllers/dev/CloudStorageStrategy.java

```diff
@@ -0,0 +1,19 @@
+package controllers.dev;
+
+import play.mvc.Http.Request;
+import play.mvc.Result;
+import repository.StoredFileRepository;
+
+/**
+ * Interface for interacting with the different cloud storage providers. Each cloud storage provider
+ * (currently we support Azure and AWS) implements the interface methods, which can then be called
+ * by FileUploadController.
+ */
+public interface CloudStorageStrategy {
```

Comment:
suggestion (blocking): these strategies are in `app/controllers/dev/` but aren't controllers and are not specific to a development task. There's not an immediately obvious place for them. One option could be to consolidate file upload code under `app/services/files`

Reply:
Ah actually `app/services/cloud` looks like the right place for this.

Reply:
Didn't realize these strategies were for Azurite and Localstack. Please clarify that in comments and ignore my previous suggestion!